### PR TITLE
Adding agentic skills for code reviews

### PR DESCRIPTION
**Pull request by shibu-kv**

### CODE REVIEW

File: .github/skills/code-review/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
+---
+name: code-review
+description: 'Analyze GitHub PR changes for embedded C/C++ code and generate comprehensive REVIEW.md with visual diffs, impact assessment, and regression risk analysis. Use when: reviewing pull requests, analyzing code changes, identifying functional regressions, assessing memory/thread safety impact, validating embedded systems changes.'
+argument-hint: 'PR URL or number (e.g., #123 or https://github.com/rdkcentral/telemetry/pull/123)'
+---
```
[the remark this answers is not in the document]
Comment (hpatel006c) [reply .github/skills/code-review/SKILL.md:5]: @copilot  Implement this suggestion.

**Comment by shibu-kv** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/rdkcentral/telemetry/pull/323#pullrequestreview-4090611712)

File: .github/skills/code-review/references/thread-patterns.md [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+## Pattern 10: Thread-Local Storage
+
+### ✅ CORRECT
+```c
+__thread char error_buffer[256];  // Each thread has its own
+
+void set_error(const char* msg) {
+    strncpy(error_buffer, msg, sizeof(error_buffer) - 1);
+    error_buffer[sizeof(error_buffer) - 1] = '\0';
```
[the remark this answers is not in the document]
Comment (shibu-kv) [reply .github/skills/code-review/references/thread-patterns.md:282]: @copilot apply changes based on [this feedback](https://github.com/rdkcentral/telemetry/pull/323#discussion_r3065132026)

File: .github/skills/code-review/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+#### Test Files (*.cpp, test/*)
+1. Test coverage → adequate for changes
+2. Mock usage → appropriate
+3. Edge cases covered
+4. Negative tests included
+
+### Step 5: Assess Regression Risk
+
+Calculate risk score based on:
```
[the remark this answers is not in the document]
Comment (shibu-kv) [reply .github/skills/code-review/SKILL.md]: @copilot apply changes based on [this feedback](https://github.com/rdkcentral/telemetry/pull/323#discussion_r3066622602)

File: .github/skills/code-review/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+Retrieve PR metadata using the GitHub CLI or API:
+
+```bash
+# PR details (title, description, author, reviewers, CI status)
+gh pr view <PR_NUMBER_OR_URL> --json number,title,body,author,reviewRequests,statusCheckRollup
+
+# File change list with line counts
+gh pr view <PR_NUMBER_OR_URL> --json files
+
+# Existing review comments
```
[the remark this answers is not in the document]
Comment (shibu-kv) [reply .github/skills/code-review/SKILL.md:110]: @copilot apply changes based on [this feedback](https://github.com/rdkcentral/telemetry/pull/323#discussion_r3066622640)

File: .github/skills/code-review/references/thread-patterns.md [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+}
+```
+
+### ✅ CORRECT (POSIX — maximum portability for pre-C11 and embedded toolchains)
+```c
+#include <pthread.h>
+
+static pthread_key_t  error_key;
+static pthread_once_t key_once = PTHREAD_ONCE_INIT;
+
```
[the remark this answers is not in the document]
Comment (shibu-kv) [reply .github/skills/code-review/references/thread-patterns.md:296]: @copilot apply changes based on [this feedback](https://github.com/rdkcentral/telemetry/pull/323#discussion_r3066622696)

**Comment by shibu-kv** [comment]
No actual source code changes. Expanding the skills.
